File: py_solutions_11-20/Euler_15.py
# find all paths in 20 X 20 grid

from __future__ import print_function


from math import factorial
import timeit


start = timeit.default_timer()

def euler_15():
    return (factorial(40) / (factorial(20) * factorial(20)))

print("Answer: {}".format(euler_15()))
stop = timeit.default_timer()
print("Time: {0:9.5f}".format(stop - start))

File: Java/java_solutions_11-20/Euler_15.java
/*
/   Project Euler problem #15 Lattice paths
/
/   How many routes are there through a 20 x 20 grid?
/
*/

import java.math.BigInteger;


public class Euler_15
{
    public static void main(String[] args)
    {
        long start = System.nanoTime();
        BigInteger factor;
        BigInteger twentySide = new BigInteger("1");
        BigInteger fortySide = new BigInteger("1");
        for (int i = 1; i <= 40; i++) {
            factor = new BigInteger(String.valueOf(i));
            fortySide = fortySide.multiply(factor);
            if (i == 20)
                twentySide = fortySide;
        }

        twentySide = twentySide.multiply(twentySide);
        fortySide = fortySide.divide(twentySide);
        long stop = System.nanoTime();
        System.out.println("Answer: " + fortySide.toString());
        System.out.printf("Time: %.4f\n", ((float) stop - start) / 1_000_000_000);
    }
}
